### PR TITLE
Fix model inference tests

### PR DESCRIPTION
**Pull request by koparasy**
This commit properly propagates compilation macro definitions to the tests.

### CODE REVIEW

File: src/ml/surrogate.hpp [hunk 1]
```diff
@@ -147,7 +147,7 @@ PERFFASPECT()
   inline void _evaluate(long num_elements,
                         long num_in,
                         size_t num_out,
-                        const TypeInValue** inputs,
+                        TypeInValue** inputs,
```
Comment (ggeorgakoudis) [inline src/ml/surrogate.hpp]: Why is const dropped?

File: src/ml/surrogate.hpp [hunk 2]
```diff
@@ -212,7 +212,7 @@ PERFFASPECT()
     _evaluate(num_elements,
               inputs.size(),
               outputs.size(),
-              static_cast<const TypeInValue**>(inputs.data()),
+              const_cast<TypeInValue**>(inputs.data()),
```
Comment (ggeorgakoudis) [inline src/ml/surrogate.hpp]: Related to comment above

File: src/ml/surrogate.hpp (hunk 1 above)
Comment (koparasy) [reply src/ml/surrogate.hpp]: I forced pushed another version and made it more coherent, without requiring changing the library header file. Resource Manager that calls umpire::deallocate seems to fail when called with constant pointers.

**Review by ggeorgakoudis — APPROVED** [review]
LGTM

**Pull request merged**